### DISCUSSION TITLE
How to create a reusable group of atoms? like mobx classes

### DISCUSSION THREAD

**Discussion by MiniSuperDev** [pmndrs/jotai#2146] [Q&A]
https://codesandbox.io/s/jotai-7guis-task4-timer-qt0o4?file=/src/atoms.ts

For example for the previous timer, I want to create multiple instances of the same group of atoms.

Reuse the logic of all the group of atoms that compose the timer, and reuse it in different apps, and I want to create different UIs for each group that I created, so reuse the state logic.

In mobx I simply create a class with this functionality inside the class, so if I want a new timer, I create a new instance, and I instantiated in the component, and that is all.

Like this
https://codesandbox.io/s/mobx-timer-tutorial-i2ylx?file=/TimerStore.js

How can i do it with jotai?

Thanks

**Comment by AjaxSolutions** [comment]
Can you just use a function which returns an atom?  Unless the function is memoized, each time the function is called it will create and return a new atom.

Alos, see this [tip:](https://twitter.com/dai_shi/status/1702511436373872783)

**Comment by wootra** [comment]
sounds like this question is just an architecture issue. as @AjaxSolutions mentioned, when atom is created again, the data structure will be reset. 
you can make this structure.
1. a function that creates an object that holds multiple atoms ( i.e. { atom1, atom2 } )
2. create an atom that calls the function for the initial value. when you re-set this atom by calling the function `#1`, this atom and all the atoms that this function returns will be updated. then the reset is automatically happens. 
3. as long as you hold the function `#1`, you can still reuse all the logic. 
I have changed your code a little to demonstrate the concept ( code is a little messy ): https://codesandbox.io/s/jotai-7guis-task4-timer-forked-5m24p5?file=/src/atoms_with_interval.ts
